File: services/admin-api/models/usage_enquiry.py
"""
Usage enquiry models for tracking demo usage requests.
"""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, EmailStr


class UsageEnquiryCreate(BaseModel):
    """Model for creating a usage enquiry."""
    name: str = ""
    email: str = ""
    company: str = ""
    role: str = ""
    solution_id: str
    solution_name: str
    skipped: bool = False


class UsageEnquiryInDB(BaseModel):
    """Model for usage enquiry stored in database."""
    enquiry_id: str
    name: str = ""
    email: str = ""
    company: str = ""
    role: str = ""
    solution_id: str
    solution_name: str
    skipped: bool = False
    created_at: datetime
    ip_address: Optional[str] = None
    user_agent: Optional[str] = None


class UsageEnquiryResponse(BaseModel):
    """Response model for a usage enquiry."""
    enquiry_id: str
    message: str

File: apps/web/src/services/usageEnquiry.ts
/**
 * Usage Enquiry API Service
 * Handles submission of usage enquiries for demo tracking.
 */

export interface UsageEnquiryData {
  name: string;
  email: string;
  company: string;
  role: string;
  solution_id: string;
  solution_name: string;
  skipped?: boolean;
}

export interface UsageEnquiryResponse {
  enquiry_id: string;
  message: string;
}

const API_BASE_URL = '/api/admin/public';

/**
 * Submit a usage enquiry to track demo usage.
 * @param data - The enquiry data to submit
 * @returns The response from the API
 */
export async function submitUsageEnquiry(
  data: UsageEnquiryData
): Promise<UsageEnquiryResponse> {
  const response = await fetch(`${API_BASE_URL}/usage-enquiry`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  if (!response.ok) {
    throw new Error(`Failed to submit enquiry: ${response.status}`);
  }

  return response.json();
}
